contracts/lib/validators.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+/// @notice Thrown if the supplied address is a zero address where it is not allowed
+error ZeroAddressNotAllowed();
+
+/// @notice Checks if the provided address is nonzero, reverts otherwise
+/// @param address_ Address to check
+/// @custom:error ZeroAddressNotAllowed is thrown if the provided address is a zero address
+function ensureNonzeroAddress(address address_) pure {
+    if (address_ == address(0)) {
+        revert ZeroAddressNotAllowed();
+    }
```

Comment:
assembly {
        extSize := extcodesize(address_)
}
we can use assembly to check zero address to save some gas.

Reply:
But it checks that there's a deployed contract, not that it's a nonzero address. Not sure if this is what we need, e.g. it would break our plan to set shortfall=0x00...01 at the launch time 😂

Reply:
My bad :disappointed: